src/borg/archiver.py

```diff
@@ -453,6 +455,14 @@ def item_is_hardlink_master(item):
                     item.get('hardlink_master', True) and 'source' not in item)
 
         filter = self.build_filter(matcher, item_is_hardlink_master, strip_components)
+        if progress:
+            logger = logging.getLogger(ProgressIndicatorPercent.LOGGER)
+            logger.info('Calculating size')
+            extracted_size = sum(item.file_size() for item in archive.iter_items(filter))
+            pi = ProgressIndicatorPercent(total=extracted_size, msg='Extracting files %5.1f%%', step=0.1)
+        else:
```

Comment:
In principle yes, but in this case I don't think it's a good idea. It saves two `if pi:` instances, but changes the parameter into "well uh sometimes it's needed, sometimes it ain't". I could change that but then it's strange semantically the other way around, requiring a parameter that may/may not be used etc.